File: one.ts
function findSubstrings(s: string): string[] {
    const result: string[] = [];
    let i = 0;

    while (i < s.length) {
        if (s[i] === '1') {
            const start = i;
            i++;

            if (s[i] === '1') i++;
            if (s[i-1] === '1' && s[i] === '0') {
                while (s[i] === '0') i++;
                if (s[i] === '1') {
                    result.push(s.slice(start, i + 1));
                    i++;
                    continue;
                }
            }
        }
        i++;
    }
    
    return result;
}

function getUserInput(): Promise<string> {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    return new Promise(resolve => {
        rl.question('Введите строку: ', (input: string) => {
            rl.close();
            resolve(input || '');
        });
    });
}

// Основная функция
async function main() {
    try {
        const inputString = await getUserInput();
        console.log('Введенная строка:', inputString);
        
        const matches = findSubstrings(inputString);
        console.log('Найденные подстроки:', matches.length > 0 ? matches : 'не найдены');
    } catch (err) {
        console.error('Произошла ошибка:', err);
    }
}

// Запуск программы
main();

File: one.kt
fun findSubstrings(input: String): List<String> {
    val result = mutableListOf<String>()
    var i = 0
    val n = input.length
    
    while (i < n - 2) {
        if (input[i] == '1') {
            if (i + 3 < n && input[i+1] == '1') {
                var j = i + 2
                while (j < n && input[j] == '0') {
                    j++
                }
                if (j > i + 2 && j < n && input[j] == '1') {
                    val substring = input.substring(i, j + 1)
                    result.add(substring)
                    i = j + 1
                    continue
                }
            }
            if (i + 2 < n && input[i+1] == '0') {
                var j = i + 1
                while (j < n && input[j] == '0') {
                    j++
                }

                if (j > i + 1 && j < n && input[j] == '1') {
                    val substring = input.substring(i, j + 1)
                    result.add(substring)
                    i = j + 1
                    continue
                }
            }
        }
        i++
    }
    
    return result
}

fun main() {
    println("Введите строку:")
    val input = readLine() ?: ""
    
    val result = findSubstrings(input)
    
    println("Результат:")
    println(result.joinToString(", "))
}